src/umpire/alloc/MallocAllocator.hpp

```diff
@@ -50,6 +58,35 @@ struct MallocAllocator {
     UMPIRE_LOG(Debug, "(ptr=" << ptr << ")");
     ::free(ptr);
   }
+
+  bool isHostPageable()
+  {
+#if defined(UMPIRE_ENABLE_CUDA)
+    int pageableMem = 0;
+    int cdev = 0;
+    cudaGetDevice(&cdev);
+
+    //Device supports coherently accessing pageable memory
+    //without calling cudaHostRegister on it
+    cudaDeviceGetAttribute(&pageableMem,
+              cudaDevAttrPageableMemoryAccess, cdev);
+    if(pageableMem)
+      return true;
+    else
+      return false;
+#endif
+    return false; //shouldn't reach this
```

Comment:
You could actually drop the "else return false" from lines 75 and 76 above and the code would just fall through to this line and return false if not pageable or UMPIRE_ENABLE_CUDA is not defined.  Then you could also remove the comment and not have to worry about knowing that the only time this function is called is if the platform is Platform::cuda.

Reply:
good idea, done!